pandas/tests/arrays/categorical/test_dtypes.py

```diff
@@ -134,3 +136,11 @@ def test_iter_python_types_datetime(self):
         cat = Categorical([Timestamp("2017-01-01"), Timestamp("2017-01-02")])
         assert isinstance(list(cat)[0], Timestamp)
         assert isinstance(cat.tolist()[0], Timestamp)
+
+    def test_interval_index_category(self):
+        # GH 38316
+        index = IntervalIndex.from_breaks(np.arange(5, dtype="uint64"))
+
+        result_subtype = CategoricalIndex(index).dtype.categories.dtype.subtype
+        expected_subtype = "uint64"
```

Comment:
Could you instead compare the result of `CategoricalIndex(index).dtype.categories` (the `IntervalIndex`)?

Reply:
Sure, Will update